app/src/test/java/doit/study/droid/data/QuizRepositoryTest.kt

```diff
@@ -0,0 +1,126 @@
+package doit.study.droid.data
+
+import CoroutineTestRule
+import androidx.room.withTransaction
+import doit.study.droid.R
+import doit.study.droid.data.local.QuizContentVersion
+import doit.study.droid.data.local.QuizDatabase
+import doit.study.droid.data.remote.Configuration
+import doit.study.droid.data.remote.QuizData
+import doit.study.droid.data.remote.QuizDataClient
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import runBlocking
+
+@ExperimentalCoroutinesApi
+class QuizRepositoryTest {
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    var coroutineRule = CoroutineTestRule()
+
+    private val quizDatabase = mockk<QuizDatabase>()
+    private val quizContentVersion = mockk<QuizContentVersion>()
+    private val quizDataClient = mockk<QuizDataClient>()
+
+    private val coroutineDispatchers = CoroutineDispatchers(
+            coroutineRule.testDispatcher,
+            coroutineRule.testDispatcher,
+            coroutineRule.testDispatcher
+    )
+
+    private val quizRepository = QuizRepository(
+            quizDatabase = quizDatabase,
+            quizContentVersion = quizContentVersion,
+            quizDataClient = quizDataClient,
+            coroutineDispatchers = coroutineDispatchers
+    )
+
+    private val fakeQuizDataItem = QuizData(
+            id = 1,
+            text = "text",
+            docRef = "doc",
+            rightAnswers = emptyList(),
+            wrongAnswers = emptyList(),
+            tags = listOf("tag")
+    )
+
+    @Before
```

Comment:
It's a bad idea to use **Before** methods and fields in test class, because each test should be as one scope. You shouldn't scroll test to find what initialized something before. Instead use high ordered functions in each test to init mock data. It's better because you can understand test easier - just at first view.